packages/firebase/src/index.ts

```diff
@@ -1,283 +1,243 @@
-import type firebase from "firebase"
-import { createHash, randomBytes } from "crypto"
-import { Adapter } from "next-auth/adapters"
-import {
-  querySnapshotToObject,
-  docSnapshotToObject,
-  stripUndefined,
-} from "./utils"
-import { Profile, Session, User } from "next-auth"
-
-interface FirebaseVerificationRequest {
-  id: string
-  identifier: string
-  token: string
-  expires: Date
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import type {
+  runTransaction,
+  collection,
+  query,
+  getDocs,
+  where,
+  limit,
+  doc,
+  getDoc,
+  addDoc,
+  updateDoc,
+  deleteDoc,
+  Firestore,
+  FirestoreDataConverter,
+} from "firebase/firestore"
+
+import type { Account } from "next-auth"
+
+import type {
+  Adapter,
+  AdapterSession,
+  AdapterUser,
+  VerificationToken,
+} from "next-auth/adapters"
+
+export const collections = {
+  Users: "users",
+  Sessions: "sessions",
+  Accounts: "accounts",
+  VerificationTokens: "verificationTokens",
+} as const
+
+export const format: FirestoreDataConverter<any> = {
+  toFirestore(object) {
+    const newObjectobject: any = {}
+    for (const key in object) {
+      const value = object[key]
+      if (value === undefined) continue
+      newObjectobject[key] = value
+    }
+    return newObjectobject
+  },
+  fromFirestore(snapshot) {
+    if (!snapshot.exists()) return null
+    const newUser: any = { ...snapshot.data(), id: snapshot.id }
+    for (const key in newUser) {
+      const value = newUser[key]
+      if (value.toDate) newUser[key] = value.toDate()
```

Comment:
```suggestion
      if (value && value.toDate) newUser[key] = value.toDate()
```
or even better
```suggestion
      if (value?.toDate) newUser[key] = value.toDate()
```

Reply:
had the same issue with firebase v9. this fix works great.

Reply:
Thank you for your suggestion! Awesome!

Reply:
yes, this solve the issue and it work in localhost. However, when i tried to deploy to Vercel, it could be build successfully, but it would have problem when login with google. When login with the google, it always show callback error, but in the localhost environmnet, there is no issue. is threre anything i should concern?